### ISSUE TITLE
[FR] Make `django.contrib.auth.base_user._UserModel` part of a public API

### ISSUE THREAD

**Issue by sterliakov** [typeddjango/django-stubs#2366]
Introduced in #2335, `_UserModel` type alias is resolved by plugin to point to current `AUTH_USER_MODEL`.

This functionality is extremely helpful in certain circumstances, esp. when typing django-related packages. I was trying to improve type hints of [djangorestframework-simplejwt](https://github.com/jazzband/djangorestframework-simplejwt/issues/787) and noticed that it'd be very convenient to use this `_UserModel` alias to spell "type of current user model". 

For example, this will help me annotate the authentication class correctly: it calls `get_user_model` internally, assigning returned class to an instance variable. Annotating with `_UserModel` would greatly improve usability compared to `AbstractBaseUser`.

I propose to add an alias to `django_stubs_ext` resolved in the same way as `_UserModel`.

I'm willing to open a PR with this change if that sounds reasonable.

**Comment by flaeppe** [comment]
Yes I agree, we should make it public. Having `_UserModel` public should be very helpful for 3rd party code.

We've previously mentioned this in: https://github.com/typeddjango/djangorestframework-stubs/issues/650#issuecomment-2285413166 and https://github.com/typeddjango/djangorestframework-stubs/pull/651#pullrequestreview-2234954083
